### PR TITLE
feat: initial implementation of the doc generator TAROT-2351

### PR DESCRIPTION
**Pull request by mrfyda**

### CODE REVIEW

File: docs/tool-description.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Semgrep accelerates your security journey by swiftly scanning code for known issues and software vulnerabilities with unparalleled efficiency. [Learn more](https://github.com/semgrep/semgrep)
```
Comment (mrfyda) [inline docs/tool-description.md:1]: @claudiacodacy @nicklem can you have a look at this one?

this is the string that will be displayed in Code Patterns page

Comment (claudiacodacy) [reply docs/tool-description.md:1]: looks good to me 👍

Comment (nicklem) [reply docs/tool-description.md:1]: It reads more commercial than other tool descriptions, but it's fitting, given the circumstances. LGTM 👍

File: internal/docgen/parsing_test.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
+package docgen
+
+import "testing"
+
+func TestPrefixRuleIDWithPath(t *testing.T) {
```
Comment (Afonsobli) [inline internal/docgen/parsing_test.go:5]: suggestion (non-blocking): should we change these variables' names? (tt, s, name, want)

File: internal/docgen/pattern-with-explanation.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+		Title:       r.Title,
+	}
+}
+
+type PatternsWithExplanation []PatternWithExplanation
+
+func (pwes PatternsWithExplanation) toCodacyPattern() []codacy.Pattern {
+	codacyPatterns := make([]codacy.Pattern, len(pwes))
+
+	for i, r := range pwes {
```
Comment (Afonsobli) [inline internal/docgen/pattern-with-explanation.go:45]: suggestion (non-blocking): change the r variable to "pattern"

**Review by Afonsobli — APPROVED** [review]

**Review by Afonsobli — APPROVED** [review]

**Pull request merged**